### DISCUSSION TITLE
About the hardcoded port 9515

### DISCUSSION THREAD

**Discussion by RoyBellingan** [php-webdriver/php-webdriver#1068] [Ideas]
So port is 9515 is just an arbitrary port like another... If is used by another process, php-webdriver can not start and we have to... manually replace it.

Can we just do something like this in https://github.com/php-webdriver/php-webdriver/blob/25a5655ea2dc63aed314535d637f83f66a984e67/lib/Firefox/FirefoxDriverService.php#L29

$port  = rand(9010,65535);

(which is what I am doing just now, and I can start multiple instance at the same time from different php process)

we could be fancy and probe if the port is open or not... 

if there is interest I can also allow the port to be configurable from outside the class to have a specific value.

**Comment by andrewnicols** [comment]
php-webdriver is a client. It calls the chromedriver client on port 9515.

You can run chromedriver on other ports. There’s an argument to set the
port, and related configuration in PHP-webdriver

**Comment by RoyBellingan** [comment]
If you check the code in [Service.php](https://github.com/php-webdriver/php-webdriver/blob/25a5655ea2dc63aed314535d637f83f66a984e67/lib/Firefox/FirefoxDriverService.php#L29)
Such argument is ignored and forcefully overriden, that is the problem.

You can not invoke chome/firefox using 
        $this->driver = FirefoxDriver::start();
        
As the port parameter will be ignored.
